### PR TITLE
Updated InfinityLib, Improvements to structure format

### PR DESCRIPTION
**Pull request by Mooy1**
## Changes
updated InfinityLib methods, working on improving structure format

## Checklist

- [ ] I have fully tested the proposed changes and promise that they will not break world generation, mob spawning, and the like.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

**Comment by Mooy1** [comment]
not quite done but ready for review

**Review by Seggan — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/addoncommunity/galactifun/core/structures/blocks/StructureBlock.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
+public class StructureBlock {
+
+    public static final StructureBlock AIR = new StructureBlock(Material.AIR);
+    private static final EnumMap<Material, StructureBlock> CACHED = new EnumMap<>(Material.class);
+
+    public static StructureBlock get(Material material) {
+        return CACHED.computeIfAbsent(material, StructureBlock::new);
+    }
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/core/structures/blocks/StructureBlock.java:27]: donnt you think "cache" is better than cached? just my opinion

File: src/main/java/io/github/addoncommunity/galactifun/Galactifun.java [hunk 2]
```diff
@@ -70,6 +77,22 @@ public void onDisable() {
...
+        // disable ender dragon spawn in end environment alien worlds
+        if (PaperLib.isPaper()) {
+            YamlConfiguration paper = new YamlConfiguration();
+            try {
+                paper.load(new File(Bukkit.getWorldContainer(), "paper.yml"));
+                paper.set("game-mechanics.scan-for-legacy-ender-dragon", false);
+            } catch (IOException | InvalidConfigurationException e) {
+                e.printStackTrace();
+            }
+        }
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/Galactifun.java:93]: dont you have to save the config?

Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/Galactifun.java:93]: o ya

**Review by Seggan — APPROVED** [review]

**Pull request merged**